pandas/core/dtypes/dtypes.py

```diff
@@ -2277,18 +2277,6 @@ def name(self) -> str:  # type: ignore[override]
     @cache_readonly
     def numpy_dtype(self) -> np.dtype:
         """Return an instance of the related numpy dtype"""
-        if pa.types.is_timestamp(self.pyarrow_dtype):
-            # pa.timestamp(unit).to_pandas_dtype() returns ns units
-            # regardless of the pyarrow timestamp units.
-            # This can be removed if/when pyarrow addresses it:
-            # https://github.com/apache/arrow/issues/34462
-            return np.dtype(f"datetime64[{self.pyarrow_dtype.unit}]")
-        if pa.types.is_duration(self.pyarrow_dtype):
-            # pa.duration(unit).to_pandas_dtype() returns ns units
-            # regardless of the pyarrow duration units
-            # This can be removed if/when pyarrow addresses it:
-            # https://github.com/apache/arrow/issues/34462
-            return np.dtype(f"timedelta64[{self.pyarrow_dtype.unit}]")
```

Comment:
By removing this, it will go through `np.dtype(self.pyarrow_dtype.to_pandas_dtype())`, which I think will raise a type error? (because `to_pandas_dtype` returns a DatetimeTZDtype, and `np.dtype(..)` does not recognize that) And so this will start to return object dtype instead of datetime64?

Reply:
For a bit of context from my linked issue, in pandas 2.0, an ArrowDtype for a pyarrow timestamp with a non-null timezone did return a numpy object dtype

The `if` statements that are removed in this pull request were added in pandas 2.1.0rc0 (https://github.com/pandas-dev/pandas/pull/51800) to fix the other issue you pointed out with the datetime unit being lost in older pyarrow versions (fix for that was added in pyarrow 14 https://github.com/apache/arrow/pull/35656). It wasn't noted in the pull request that added them that it was intended to change the semantics for tz-aware types as well, so I think it was just an unintentional side effect that they started returning a numpy dtype of datetime64 instead of object.

I noted in the issue that [the pyarrow table](https://pandas.pydata.org/docs/reference/arrays.html#pyarrow) in the pandas arrays, scalars, and data types section of the API docs seems to indicate that a pyarrow timestamp _should_ map to a pandas DatetimeTZDtype _and_ a numpy datetime64 dtype (which does match how pyarrow itself handles converting a tz-aware array to pandas and numpy respectively)

Quoting from my comment on the issue
> I would definitely defer to someone else's judgement on whether that is correct, or if there should be a distinction in that table linked between a pa.timestamp() type with and without timezone

You are pretty much exactly the person I had in mind of who would be best suited to make that judgement call. As far as I can tell, a tz-aware pyarrow timestamp is the only instance in that table that can lose information when it maps to the shown numpy dtype